components/style/properties/shorthand/border.mako.rs

```diff
@@ -132,14 +132,34 @@ pub fn parse_border(context: &ParserContext, input: &mut Parser)
 
     impl<'a> LonghandsToSerialize<'a>  {
         fn to_css_declared<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
-            // If all longhands are all present, then all sides should be the same,
-            // so we can just one set of color/style/width
-            super::serialize_directional_border(
-                dest,
-                self.border_${side}_width,
-                self.border_${side}_style,
-                self.border_${side}_color
-            )
+            % for side in PHYSICAL_SIDES:
+              let border_${side}_width = self.border_${side}_width.clone();
```

Comment:
are the clone needed? I'm pretty sure you can avoid it using something like:

```rust
let all_equal = {
    % for ...:
        let border_${...}_style = &self.border_${...}_style;
    % endfor
    border_top_width == border_right_width &&
    // ...
};
```

Or just using self everywhere, whatever you prefer.